android/src/main/java/com/plaid/PLKEmbeddedView.kt

```diff
@@ -0,0 +1,161 @@
+package com.plaid
+
+import android.app.Application
+import android.content.Context
+import android.content.Intent
+import android.util.AttributeSet
+import android.util.Log
+import android.view.View
+import android.widget.FrameLayout
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.bridge.NativeModule
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.uimanager.ThemedReactContext
+import com.facebook.react.uimanager.events.RCTEventEmitter
+import com.plaid.gson.PlaidJsonConverter
+import com.plaid.link.OpenPlaidLink
+import com.plaid.link.Plaid.create
+import com.plaid.link.Plaid.createLinkEmbeddedView
+import com.plaid.link.Plaid.setLinkEventListener
+import com.plaid.link.configuration.LinkTokenConfiguration
+import com.plaid.link.configuration.LinkTokenConfiguration.Builder
+import com.plaid.link.event.LinkEvent
+import com.plaid.link.result.LinkExit
+import com.plaid.link.result.LinkSuccess
+import org.json.JSONException
+import org.json.JSONObject
+
+class PLKEmbeddedView @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null
+) : FrameLayout(context, attrs), ActivityResultHandler {
+  private val themedReactContext: ThemedReactContext = context as ThemedReactContext
+  private val TAG = "EmbeddedSearch"
+  private val LINK_ACTIVITY_REQUEST_CODE = 3364
+  private val EVENT_NAME = "OnEmbeddedEvent"
+  private val jsonConverter = PlaidJsonConverter()
+
+  init {
+    inflate(context, R.layout.plk_embedded_view, this)
+    setupOnEventListener()
+  }
+
+  fun setToken(token: String) {
```

Comment:
@dtroupe-plaid Could this field be nullable?

Reply:
I don't think it will be an issue, but we should test it.